vignettes/pandoc_table.Rmd

```diff
@@ -0,0 +1,195 @@
+---
+title: "Rendering tables with pandoc.table"
+author: "Roman Tsegelskyi "
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Pandoc.table}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, echo = FALSE, message = FALSE}
+knitr::opts_chunk$set(collapse = T, comment = "#>")
+library(pander)
+```
+
+Core functionality of `pander` is centered around `pandoc.table` which is aimed at rendering tables in `markdown`. In case of 2D tables, `pander` calls `pandoc.table` internally, thus in such cases `pander` and `pandoc.table` support the same argument and in this vignette will be used iterchangingly. `pandoc.table` has big variety of options ([highlighting](#highlighting-cells), [styles](#table-styles), etc.) and this vignette aims to give a more details overview of most common options.
+
+## Table styles
+
+Since `pander` aims at rendering R objects into [Pandoc](http://pandoc.org/)'s markdown all four (`multiline`, `simple`, `grid`, `rmarkdown`) of  [Pandoc](http://johnmacfarlane.net/pandoc/README.html#tables)'s formats are supported. General advices is to stick with default `multiline` style, but if there is a need to change it either specify `style` argument when calling `pander/pandoc.table` or change default `style` using `panderOptions`.
+
+### multiline tables
+
+`multiline` tables allow headers and table rows to span multiple lines of text (but cells that span multiple columns or rows of the table are not supported). Also note, that, by default, line breaks are removed from cells for simplicity, so in order to have multiline cells `keep.line.breaks` needs to be set to `TRUE`:
+
+```{r}
+m <- data.frame('Value\n1', 'Value\n2')
+colnames(m) <- c('Multiline\nCol1', 'Multiline\nCol2')
+pandoc.table(m, keep.line.breaks = TRUE)
+m <- mtcars[1:3, 1:4]
+pandoc.table(m)
+```
+
+### simple tables
+
+`simple` tables are have more compact syntax that all other styles, but don't they support multiline cells:
+
+```{r, error=TRUE}
+m <- mtcars[1:3, 1:4]
+pandoc.table(m, style = 'simple')
+m <- data.frame('Value\n1', 'Value\n2')
+colnames(m) <- c('Multiline\nCol1', 'Multiline\nCol2')
+pandoc.table(m, keep.line.breaks = TRUE, style='simple')
+```
+
+### grid tables
+
+`grid` format is really handy for `emacs` users ([Emacs table mod](http://table.sourceforge.net/)) and it does support block elements (multiple paragraphs, code blocks, lists, etc.) inside cells. Alignments are not supported by most parsers, meaning that even though `pander` will produce a table with alightnment, it will be lost during conversion from `markdown` to `HTML/PDF/DOCX`, and cells can't span multiple columns or rows.
```

Comment:
But `pandoc` handles alignment very well -- and probably most `pander` users are using `pandoc` anyway (even if they do not know that -- via RStudio).